packages/in_app_purchase/lib/src/google_play_connection.dart

```diff
@@ -0,0 +1,33 @@
+import 'dart:async';
+
+import 'billing_client_wrappers.dart';
+import 'in_app_purchase_connection.dart';
+
+/// An [InAppPurchaseConnection] that wraps Google Play Billing.
+///
+/// This translates various [BillingClient] calls and responses into the
+/// common plugin API.
+class GooglePlayConnection implements InAppPurchaseConnection {
+  BillingClient _billingClient = BillingClient();
+
+  @override
+  Future<bool> isAvailable() async {
+    return await _billingClient.isReady;
+  }
+
+  @override
+  Future<bool> connect() async {
+    final BillingResponse responseCode = await _billingClient.startConnection(
+        onBillingServiceDisconnected: () {});
+    final bool finishedSuccessfully = responseCode == BillingResponse.OK;
+    if (!finishedSuccessfully) {
+      print('Failed to connect to Play Billing. Response $responseCode');
```

Comment:
We probably want to expose some information about why we couldn't connect to the appm e.g if there's no internet connection (which I assume is a requirement), the app might want to say something to the user.